app/edge/views/edge-stacks/createEdgeStackView/create-edge-stack-view.controller.js

```diff
@@ -363,12 +367,33 @@ export default class CreateEdgeStackViewController {
     });
   }
 
+  validateEnvVars(envVars) {
+    for (const obj of envVars) {
+      for (const key in obj) {
+        if (obj[key] === '') {
+          return false;
+        }
+      }
+    }
+    return true;
```

Comment:
```suggestion
    return envVars.every(({name, value}) => !!name && !!value)
```

that's equivalent, but IMO clearer to what the check does. 

the problem with both is that an env var can have an empty value, but the app templates variables can't. 

IMO you should have a check on `this.state.templateValues.envVars`. sthing like
```ts
Object.entries(this.state.templateValues.envVars).every(([, value]) => !!value)
```

or better: (using app/react/components/form-components/validate-form.ts and app/react/edge/edge-stacks/CreateView/TemplateFieldset/EnvVarsFieldset.tsx)
```ts
const errors = await validateForm(envVarsFieldsetValidation(), this.state.templateValues.envVars)
return Object.keys(errors).length >0
```

maybe add the same for custom template variables

Reply:
It has a schemaBuilder related error while using the `await validateForm` solution. The `Object.entries` one works well, so I used it for validating both variables and env vars.